packages/gatsby-link/src/index.js

```diff
@@ -181,70 +198,68 @@ class GatsbyLink extends React.Component {
       partiallyActive,
       state,
       replace,
+      _location,
       /* eslint-enable no-unused-vars */
       ...rest
     } = this.props
+
     if (process.env.NODE_ENV !== `production` && !isLocalLink(to)) {
       console.warn(
         `External link ${to} was detected in a Link component. Use the Link component only for internal links. See: https://gatsby.dev/internal-links`
       )
     }
 
+    const prefixedTo = rewriteLinkPath(to, _location.pathname)
+    if (!isLocalLink(prefixedTo)) {
+      return <a href={prefixedTo} {...rest} />
+    }
+
     return (
-      <Location>
-        {({ location }) => {
-          const prefixedTo = rewriteLinkPath(to, location.pathname)
-          return isLocalLink(prefixedTo) ? (
-            <Link
-              to={prefixedTo}
-              state={state}
-              getProps={getProps}
-              innerRef={this.handleRef}
-              onMouseEnter={e => {
-                if (onMouseEnter) {
-                  onMouseEnter(e)
-                }
-                ___loader.hovering(parsePath(prefixedTo).pathname)
-              }}
-              onClick={e => {
-                if (onClick) {
-                  onClick(e)
-                }
-
-                if (
-                  e.button === 0 && // ignore right clicks
-                  !this.props.target && // let browser handle "target=_blank"
-                  !e.defaultPrevented && // onClick prevented default
-                  !e.metaKey && // ignore clicks with modifier keys...
-                  !e.altKey &&
-                  !e.ctrlKey &&
-                  !e.shiftKey
-                ) {
-                  e.preventDefault()
-
-                  let shouldReplace = replace
-                  const isCurrent =
-                    encodeURI(prefixedTo) === window.location.pathname
-                  if (typeof replace !== `boolean` && isCurrent) {
-                    shouldReplace = true
-                  }
-                  // Make sure the necessary scripts and data are
-                  // loaded before continuing.
-                  window.___navigate(prefixedTo, {
-                    state,
-                    replace: shouldReplace,
-                  })
-                }
-
-                return true
-              }}
-              {...rest}
-            />
-          ) : (
-            <a href={prefixedTo} {...rest} />
-          )
+      <Link
```

Comment:
I unraveled the if and location, nothing else changed

Reply:
For sure, looks cleaner, more dry with _prefetch() and appears that it will solve the bug 😄 Woo 🎉